graylog2-web-interface/src/routing/App.jsx

```diff
@@ -1,54 +1,58 @@
 import PropTypes from 'prop-types';
 import React from 'react';
-import createReactClass from 'create-react-class';
-import Reflux from 'reflux';
+
 import Navigation from 'components/navigation/Navigation';
 import Spinner from 'components/common/Spinner';
 import Footer from 'components/layout/Footer';
+import connect from 'stores/connect';
+import StoreProvider from 'injection/StoreProvider';
+
+import AppErrorBoundary from './AppErrorBoundary';
 
 import 'stylesheets/jquery.dynatable.css';
 import 'stylesheets/typeahead.less';
 import 'c3/c3.css';
 import 'dc/dc.css';
 
-import StoreProvider from 'injection/StoreProvider';
-import AppErrorBoundary from './AppErrorBoundary';
-
 const CurrentUserStore = StoreProvider.getStore('CurrentUser');
 
-const App = createReactClass({
-  displayName: 'App',
-
-  propTypes: {
-    children: PropTypes.oneOfType([
-      PropTypes.arrayOf(PropTypes.element),
-      PropTypes.element,
-    ]).isRequired,
-    location: PropTypes.object.isRequired,
-  },
-
-  mixins: [Reflux.connect(CurrentUserStore)],
-
-  render() {
-    if (!this.state.currentUser) {
-      return <Spinner />;
-    }
-    return (
-      <div>
-        <Navigation requestPath={this.props.location.pathname}
-                    fullName={this.state.currentUser.full_name}
-                    loginName={this.state.currentUser.username}
-                    permissions={this.state.currentUser.permissions} />
-        <div id="scroll-to-hint" style={{ display: 'none' }} className="alpha80">
-          <i className="fa fa-arrow-up" />
-        </div>
-        <AppErrorBoundary>
-          {this.props.children}
-        </AppErrorBoundary>
-        <Footer />
+const App = ({ children, currentUser, location }) => {
+  if (Object.keys(currentUser).length === 0) {
```

Comment:
Just to be sure, the `currentUser: { currentUser } = {}` is always overwriting the `App.defaultProps`, right?
I think we should still get rid of the default prop `currentUser: undefined` because it would throw an error when using it with `Object.keys(currentUser).length`

Reply:
Ah, just saw dennis answer for the first part!